.github/workflows/ghcr-publish.yaml

```diff
@@ -0,0 +1,47 @@
+name: Docker Image build and deploy to GHCR
+
+on:
+  push:
+    branches:
+      - 'main'
+    tags:
+      - 'v*'
+
+env:
+  REGISTRY: ghcr.io
+  IMAGE_NAME: ${{ github.repository }}
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    
+    permissions:
+      contents: read
+      packages: write
+
+    steps:
+    - uses: actions/checkout@v3
+    
+    - name: Log in to the Container registry
+      uses: docker/login-action@v3
+      with:
+        registry: ${{ env.REGISTRY }}
+        username: ${{ github.actor }}
+        password: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
I've generated a token and set the secret but this needs to be `secrets.GHCR_TOKEN` since Github secrets cannot start with `GITHUB_` for their name.

Reply:
Done.